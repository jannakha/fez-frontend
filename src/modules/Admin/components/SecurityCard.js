import React from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

import { StandardCard } from 'modules/SharedComponents/Toolbox/StandardCard';
import SecuritySelector from './SecuritySelector';

export const SecurityCard = ({ disabled, entity, fieldID, text, selectedPolicyKey = 0, selectedDataStreamPolicyKey = 0 }) => {
    const title = (
        <span>
            <b>{entity.type}</b> level security - {entity.pid}
        </span>
    );
    return (
        <StandardCard title={title} accentHeader>
            <Grid container spacing={8}>
                <Grid item xs={12}>
                    <Typography variant="body2" component="p">
                        {text.description}
                    </Typography>
                </Grid>
                <SecuritySelector
                    disabled={disabled}
                    selectedPolicyKey={selectedPolicyKey}
                    text={text}
                    fieldID={fieldID}
                />
            </Grid>
            {
                entity.dataStreamPolicyID &&
                <Grid container spacing={8}>
                    <SecuritySelector
                        disabled={disabled}
                        selectedPolicyKey={selectedDataStreamPolicyKey}
                        text={{
                            ...text,
                            fieldLabel: text.dataStreamFieldLabel,
                            selectedTitle: text.dataStreamSelectedTitle
                        }}
                        fieldID={fieldID.replace('Security', 'DataStreamSecurity')}
                    />
                </Grid>
            }
        </StandardCard>
    );
};

SecurityCard.propTypes = {
    disabled: PropTypes.bool,
    entity: PropTypes.object,
    selectedPolicyKey: PropTypes.number,
    selectedDataStreamPolicyKey: PropTypes.number,
    text: PropTypes.object,
    fieldID: PropTypes.string,
};

export default React.memo(SecurityCard);
