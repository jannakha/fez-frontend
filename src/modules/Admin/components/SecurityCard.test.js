import { SecurityCard } from './SecurityCard';

function setup(testProps, isShallow = true) {
    const props = {
        disabled: false,
        selectedPolicyKey: 1,
        fieldID: 'test1',
        entity: {
            type: 'test2',
            dataStreamPolicyID: 2
        },
        text: {
            description: 'test3',
            prompt: 'test4'
        },
        ...testProps
    };

    return getElement(SecurityCard, props, isShallow);
}

describe('SecurityCard component', () => {

    it('should render properly', () => {
        const wrapper1 = setup({});
        expect(toJson(wrapper1)).toMatchSnapshot();

        const wrapper2 = setup({ selectedPolicyKey: undefined });
        expect(toJson(wrapper2)).toMatchSnapshot();
    });

});