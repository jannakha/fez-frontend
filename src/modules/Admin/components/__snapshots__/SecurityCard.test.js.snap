// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityCard component should render properly 1`] = `
<StandardCard
  accentHeader={true}
  title={
    <span>
      <b>
        test2
      </b>
       level security - 
    </span>
  }
>
  <WithStyles(Grid)
    container={true}
    spacing={8}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        component="p"
        variant="body2"
      >
        test3
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <[object Object]
      disabled={false}
      fieldID="test1"
      selectedPolicyKey={1}
      text={
        Object {
          "description": "test3",
          "prompt": "test4",
        }
      }
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    spacing={8}
  >
    <[object Object]
      disabled={false}
      fieldID="test1"
      selectedPolicyKey={0}
      text={
        Object {
          "description": "test3",
          "fieldLabel": undefined,
          "prompt": "test4",
          "selectedTitle": undefined,
        }
      }
    />
  </WithStyles(Grid)>
</StandardCard>
`;

exports[`SecurityCard component should render properly 2`] = `
<StandardCard
  accentHeader={true}
  title={
    <span>
      <b>
        test2
      </b>
       level security - 
    </span>
  }
>
  <WithStyles(Grid)
    container={true}
    spacing={8}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        component="p"
        variant="body2"
      >
        test3
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <[object Object]
      disabled={false}
      fieldID="test1"
      selectedPolicyKey={0}
      text={
        Object {
          "description": "test3",
          "prompt": "test4",
        }
      }
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    spacing={8}
  >
    <[object Object]
      disabled={false}
      fieldID="test1"
      selectedPolicyKey={0}
      text={
        Object {
          "description": "test3",
          "fieldLabel": undefined,
          "prompt": "test4",
          "selectedTitle": undefined,
        }
      }
    />
  </WithStyles(Grid)>
</StandardCard>
`;
